Atomic/diff.py

```diff
@@ -33,48 +72,143 @@ def diff(self):
                         raise ValueError("{0} is not RPM based.".format(rpmimage.name))
                     rpmimage._get_rpm_content()
                     rpm_image_list.append(rpmimage)
+            file_diff = DiffFS(image_list[0].chroot, image_list[1].chroot)
+            diff_dict[first_image]["first_release"] = (rpm_image_list[0]).release
+            diff_dict[second_image]["second_release"] = (rpm_image_list[1]).release
 
             if not self.args.no_files:
-                helpers.output_files(images, image_list)
+                diff_dict[first_image]["first_only"] = file_diff.left
+                diff_dict[second_image]["second_only"] = file_diff.right
+                diff_dict["Both"]["files_differ"] = file_diff.common_diff
 
             if self.args.rpms:
-                helpers.output_rpms(rpm_image_list)
+                ip = RpmPrint(rpm_image_list)
+                diff_dict[first_image]["first_rpm"] = []
+                diff_dict[second_image]["second_rpm"] = []
+                if ip.has_diff:
+                    for rpm in ip.all_rpms:
+                        if rpm in ip.i1.rpms and rpm in ip.i2.rpms:
+                            diff_dict[first_image]["first_rpm"].append(rpm)
+                            diff_dict[second_image]["second_rpm"].append(rpm)
+
+                        elif rpm in ip.i1.rpms:
+                            diff_dict[first_image]["first_rpm"].append(rpm)
+
+                        elif rpm in ip.i2.rpms:
+                            diff_dict[second_image]["second_rpm"].append(rpm)
 
-            # Clean up
             helpers._cleanup(image_list)
 
-            if self.args.json:
-                util.output_json(helpers.json_out)
+            return diff_dict
+
         except KeyboardInterrupt:
             util.write_out("Quitting...")
             helpers._cleanup(image_list)
 
+
+    def output_files(self, images, diff_dict):
+        '''
```

Comment:
Could you put output_files back to where it was originally so we get a better idea of the diffs.  Moving these methods around causes review to be too difficult.